### ISSUE TITLE
How to use a trained model?

### ISSUE THREAD

**Issue by GabrielLin** [zoranmedic/DualLCR#3]
After training, how to use the trained model. For example, given a sentence or paragraph or document to recommend citations? Thanks.

**Comment by zoranmedic** [comment]
Hi @GabrielLin,

If you want to use the pretrained model for prediction on test files from the two datasets used in the experiments, you should run the `predict.py` script. 
Here is an example for obtaining predictions with the DualEnh-ws variant:
`python predict.py <path_to_config_json> <path_to_saved_model> -dual -global_info -weighted_sum <path_to_input_file>
`

If you would like to use your own collection of sentences (i.e., citation contexts) and articles for recommendation, you should first preprocess the articles and sentences using `preprocess.py` in `datasets/` folder. After that you should update the fields in the `config.json` file so they point to the preprocessed contexts/articles files and then run the `predict.py` script with appropriate arguments (depending on the model variant) and a new input file with pairs of contexts and articles that should be scored.

Hope this helps! 
Let me know if there is something unclear,
Zoran

**Comment by GabrielLin** [comment]
@zoranmedic Thank you for your reply. 
I would like to use my own sentences but I am not sure how to construct the JSON for the parameter `input_file`. Could you please provide a sample JSON file to the `preprocess.py`. 

Moreover, there is a `-papers` parameter. I think there should be two different formats of the JSON file depending on this parameter. Thank you again.

**Comment by zoranmedic** [comment]
Hi @GabrielLin! 

Apologies for a bit late reply. 

I realised that I didn't include the method for masking citations in citation contexts in the `preprocess.py` so I've added it now. I've also included two sample JSONs in `sample_data/` folder, together with instructions on how to preprocess the data that is not from the two datasets used in the paper. 

Basically, you should make sure that your JSONs follow the format of the sample files and default preprocessing (as given in the instructions should work). However, you might need to adapt the `mask_citations` method if your input contains citations in different style than assumed in the script. Check out the defined regular expressions to see which formats are supported.

Hope this helps!

**Comment by GabrielLin** [comment]
@zoranmedic , thank you very much. You are helpful.
